src/server.ts

```diff
@@ -46,3 +50,143 @@ async function connectToDBAndStartListening() {
         );
     });
 }
+
+app.get("/:deckid/flashcards", async (req, res) => {
+    try {
+        const { deckid } = req.params;
+        const text =
+            "SELECT * FROM flashcard WHERE deck_id = $1 AND next_review <= now() ORDER BY next_review";
+        const values = [deckid];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        const deckCards = response.rows;
+        res.status(200).json(deckCards);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.get("/decks/:id", async (req, res) => {
+    try {
+        const { id } = req.params;
+        const text = "SELECT * FROM deck WHERE user_id = $1";
+        const values = [id];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        const userDecks = response.rows;
+        res.status(200).json(userDecks);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.post("/flashcards", async (req, res) => {
+    try {
+        const { front, back, deck_id } = req.body;
+        const text =
+            "INSERT INTO flashcard(front, back, deck_id) VALUES($1, $2, $3)";
+        const values = [front, back, deck_id];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.post("/decks", async (req, res) => {
+    try {
+        const { name, user_id } = req.body;
+        const text = "INSERT INTO deck(name, user_id) VALUES($1, $2)";
+        const values = [name, user_id];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.post("/users", async (req, res) => {
+    try {
+        const { user_name } = req.body;
+        const text = "INSERT INTO users(name) VALUES($1)";
+        const values = [user_name];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.get("/users", async (_req, res) => {
+    try {
+        const text = "SELECT * FROM users";
+        qNum++;
+        const response = await queryAndLog(qNum, client, text);
+        const allUsers = response.rows;
+        res.status(200).json(allUsers);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.put("/flashcards/:id", async (req, res) => {
+    try {
+        const { streak } = req.body;
+        const { id } = req.params;
+        const interval = getInterval(streak);
+        const hours = `${interval * 24} hours`;
```

Comment:
Should these be days and not hours?

Reply:
We used hours primarily because we had timezone and sql syntax problems. After fixing these problems, we didn't go back to check if days would also work. Nice catch